File: Year-2023/LeetCode-January-Challenge/DeleteColumnstoMakeSorted.js
var minDeletionSize = function(strs) {
    let res = 0;
    for(let col = 0; col < strs[0].length; ++col){
        for(let row = 1; row < strs.length; ++row){
            if(strs[row - 1][col] > strs[row][col]){
                res++;
                break;
            }
        }
    }
    return res;
};
let strs = ["cba","daf","ghi"];
console.log(minDeletionSize(strs));

File: Year-2023/LeetCode-January-Challenge/DeleteColumnstoMakeSorted.java
class Solution {
    public int minDeletionSize(String[] strs) {
        int res = 0;
        for(int col = 0; col < strs[0].length(); ++col){
            for(int row = 1; row < strs.length; ++row){
                if(strs[row - 1].charAt(col) > strs[row].charAt(col)){
                    res++;
                    break;
                }
            }
        }
        return res;
    }
}
public class DeleteColumnstoMakeSorted{
    public static void main(String[] args) {
        String[] strs = {"cba","daf","ghi"};
        Solution s = new Solution();
        System.out.println(s.minDeletionSize(strs));
    }
}